R/coerce.R

```diff
@@ -0,0 +1,180 @@
+#'   Alternatively, the \code{spatialData} argument may be provided. If both are
+#'   provided, \code{spatialDataNames} is given precedence, and a warning is
+#'   returned.
+#' @param spatialCoordsNames A \code{character} vector of column names from
+#'   \code{\link{colData}} or \code{\link{spatialData}} containing spatial
+#'   coordinates, which will be accessible with \code{\link{spatialCoords}}.
+#'   Alternatively, the \code{spatialCoords} argument may be provided. If both
+#'   are provided, \code{spatialCoordsNames} is given precedence, and a warning
+#'   is returned. Default = \code{c("x", "y")}.
+#' @param spatialData A \code{\link{DataFrame}} containing columns to store in
+#'   \code{\link{spatialData}}, which must contain at least the columns of
+#'   spatial coordinates. Alternatively, \code{spatialDataNames} may be
+#'   provided. If both are provided, \code{spatialDataNames} is given
+#'   precedence, and a warning is returned.
+#' @param spatialCoords A numeric matrix containing columns of spatial
+#'   coordinates, which will be accessible with \code{\link{spatialCoords}}.
+#'   Alternatively, \code{spatialCoordsNames} may be provided. If both are
+#'   provided, \code{spatialCoordsNames} is given precedence, and a warning is
+#'   returned.
+#' @param scaleFactors Optional scale factors associated with the image(s). This
+#'   can be provided as a numeric value, numeric vector, list, or file path to a
+#'   JSON file for the 10x Genomics Visium platform. For 10x Genomics Visium,
+#'   the correct scale factor will automatically be selected depending on the
+#'   resolution of the image from \code{imageSources}. Default = \code{1}.
+#' @param imgData Optional \code{\link{DataFrame}} containing the image data.
+#'   Alternatively, this can be built from the arguments \code{imageSources} and
+#'   \code{image_id} (see Details).
+#' @param imageSources Optional file path(s) or URL(s) for one or more image
+#'   sources.
+#' @param image_id Optional character vector (same length as
+#'   \code{imageSources}) containing unique image identifiers.
+#' @param loadImage Logical indicating whether to load image into memory.
+#'   Default = \code{FALSE}.
+
+#' @aliases 
+#' coerce, SingleCellExperiment, SpatialExperiment-method
+#' toSpatialExperiment
+#' 
+#' @examples 
+#' dir <- system.file(
+#' file.path("extdata", "10xVisium", "section1"),
+#' package = "SpatialExperiment")
+#' 
+#' # read in counts
+#' fnm <- file.path(dir, "raw_feature_bc_matrix")
+#' sce <- DropletUtils::read10xCounts(fnm)
+#' 
+#' 
+#' # read in spatial coordinates
+#' fnm <- file.path(dir, "spatial", "tissue_positions_list.csv")
+#' xyz <- read.csv(fnm, header = FALSE,
+#'                 col.names = c(
+#'                     "barcode", "in_tissue", "array_row", "array_col",
+#'                     "pxl_row_in_fullres", "pxl_col_in_fullres"))
+#' 
+#' (spe <- as(sce, "SpatialExperiment"))
+#' ## as method
```

Comment:
Maybe clearer if this comment `## as method` is above the previous line instead of after it

Reply:
done